### PR TITLE
issue #220 - re-enable JDBCSearchDateTest and fix related issues

### PR DESCRIPTION
**Pull request by lmsurpre**
Enabling this test as part of issue #92 had uncovered issue #220.
This commit addresses issue #220 by passing FHIR Elements to the JDBC
"Processor" component instead of FHIRPathNodes.

Previously, we were feeding the process method the entire FHIRPathNode
and it was processing the start and end times separately (instead of
handling them together as a date range).

Further, to make the JDBCNormSearchDateTest work, I
1. moved the BasicDate.json example to the fhir-examples project and updated it for R4
2. added a helper method in the AbstractPLSearchTest to facilitate
reading examples from fhir-examples
3. updated the extension-search-parameters.json config for the "date"
tenant and introduced the newly required
extension-search-parameters-valuetypes.json file
4. moved the `createResources` step into the setup in order to avoid
"dependsOn" and the corresponding issue I opened at
https://github.com/cbeust/testng-eclipse/issues/435 ...this change
allows us to run individual test cases from the test class whereas that
used to fail. To ensure this runs after the @BeforeClass methods being
used to set the tenantId in the concrete subclasses, I added all the
tests to a group and used the @BeforeGroups annotation; perhaps there is
a cleaner way but this seems to work ok.

I also updated the default unit test logging properties and some error handling logic to ensure we get unexpected exceptions printed to the console.

If this approach is supported by the team, I will proceed to enable the
other search tests in a similar fashion.

Signed-off-by: lmsurpre <lmsurpre@us.ibm.com>

### CODE REVIEW

**Comment by prb112** [comment]
**IT** results did not pass 2 test.  
```
[ERROR] Tests run: 311, Failures: 2, Errors: 0, Skipped: 5, Time elapsed: 32.111 s <<< FAILURE! - in TestSuite
[ERROR] processExamples(com.ibm.fhir.server.test.R4ExampleServerTest)  Time elapsed: 2.867 s  <<< FAILURE!
java.lang.AssertionError: expected:<201> but was:<500>
	at com.ibm.fhir.server.test.R4ExampleServerTest.processExamples(R4ExampleServerTest.java:31)

[ERROR] testCreatePatient(com.ibm.fhir.server.test.SearchAllTest)  Time elapsed: 0.054 s  <<< FAILURE!
java.lang.AssertionError: expected:<201> but was:<500>
	at com.ibm.fhir.server.test.SearchAllTest.testCreatePatient(SearchAllTest.java:64)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   R4ExampleServerTest.processExamples:31->FHIRServerTestBase.assertResponse:426 expected:<201> but was:<500>
[ERROR]   SearchAllTest.testCreatePatient:64->FHIRServerTestBase.assertResponse:426 expected:<201> but was:<500>
[INFO] 
[ERROR] Tests run: 311, Failures: 2, Errors: 0, Skipped: 5
```

File: fhir-persistence-jdbc/src/test/resources/config/date/extension-search-parameters.json [hunk 1]
```diff
@@ -510,7 +510,7 @@
                     "Basic"
                 ],
                 "type": "date",
-                "expression": "Basic.extension.where(url='http://example.org/missing-Period').valuePeriod",
+                "expression": "Basic.extension.where(url='http://example.org/missing-Period').value",
```
Comment (prb112) [inline fhir-persistence-jdbc/src/test/resources/config/date/extension-search-parameters.json]: I double checked these files for consistency 
./fhir-server/liberty-config/config/default/extension-search-parameters.json
./fhir-server/liberty-config-tenants/config/tenant1/extension-search-parameters.json
./fhir-server/liberty-config-tenants/config/tenant2/extension-search-parameters.json

There may be others I missed, and causing the IT failures?

**Comment by prb112** [comment]
I've reviewed the code, it looks good, the BUILD and IT failed.

**Comment by prb112** [comment]
```
 [ERROR] Tests run: 19, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 62.655 s <<< FAILURE! - in TestSuite
[ERROR] perform(com.ibm.fhir.persistence.jdbc.test.spec.R4JDBCExamplesTest)  Time elapsed: 24.429 s  <<< FAILURE!
com.ibm.fhir.persistence.exception.FHIRPersistenceProcessorException: Unexpected error while processing parameter composed-of
	at com.ibm.fhir.persistence.jdbc.test.spec.R4JDBCExamplesTest.perform(R4JDBCExamplesTest.java:54)
Caused by: java.lang.NoSuchMethodException: com.ibm.fhir.persistence.jdbc.util.JDBCParameterBuilder.process(com.ibm.fhir.model.resource.SearchParameter, com.ibm.fhir.model.type.Canonical)
	at com.ibm.fhir.persistence.jdbc.test.spec.R4JDBCExamplesTest.perform(R4JDBCExamplesTest.java:54)
```
Build failed on noSuchMethod

**Comment by lmsurpre** [comment]
> Build failed on noSuchMethod
Makes sense since the Canonical data type is new since DSTU2.  Will look for other new datatypes as well and support them in `Processor`.

**Pull request merged**